go/vt/orchestrator/inst/instance_topology_dao.go

```diff
@@ -360,7 +360,9 @@ func StopReplicas(replicas [](*Instance), stopReplicationMethod StopReplicationM
 
 // StopReplicasNicely will attemt to stop all given replicas nicely, up to timeout
 func StopReplicasNicely(replicas [](*Instance), timeout time.Duration) [](*Instance) {
-	return StopReplicas(replicas, StopReplicationNice, timeout)
+	stoppedReplicas := StopReplicas(replicas, StopReplicationNice, timeout)
+	stoppedReplicas = RemoveNilInstances(stoppedReplicas)
```

Comment:
I ran into a bug in `electNewPrimary` in VtOrc while testing my PR and this was a fix for it. StopReplias sometimes returns a nil value in the list, and we should not consider those replicas for further election

Reply:
Do we know why it could return nil values? It seems that we could be masking here another bug in a `StopReplicas`

Reply:
Over here in this line https://github.com/vitessio/vitess/blob/07881bbaddab351b518192e98e03678808fa6bd1/go/vt/orchestrator/inst/instance_topology_dao.go#L350 we are ignoring the error coming from the `StopReplication` function. That error could be coming from Reading of topology instance which will also return nil in that case. This can happen if a tablet was forgotten (removed from backend data after it went down) and we tried to read that instance. 

We only use `StopReplicas` in two places, and we had this RemoveNilInstances in the other place too. 
Also, I had asked @shlomi-noach  about this earlier. And this was the reply that he gave.

> I distinctly remember writing this change, after having seen orchestrator crash due to a nil replica. This can happen when "replica, _ = StopReplication(&replica.Key)" in StopReplicas() function fails to read the replica, because of whatever glitch.

Reply:
With the `RemoveNilInstances`, we are preventing panics from occurring in the code which is using these returned tablets

Reply:
Basically we re-read an instance after any operation that modifies its sate, and `StopReplica` is one such cases. When reading instance info we run like 10-12 queries; any single one could fail just because the world is not perfect, which generally fails the entire `ReadTopologyInstance` embedded function, which then also returns an error on your original operation (e.g. your `StopReplica`). Now, sometimes we really want to just push on, and not fail some grand operation just because some random query like `SHOW GLOBAL VARIABLES ...` had a glitch. This was one of those cases.